PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,584 @@
+            #>
+    
+            $____ | Should -Be ('pears' -match 'p*ears')
+            $____ | Should -Be ('shears' -match 'p*ears')
+        }
+
+        It 'does not need to match the entire string' {
+            <#
+                What happened in that last example? What part of 'shears' matched the pattern 
+                'p*ears'? When you use the -match operator, you can see some interesting information
+                in the automatic variable $matches. $matches is a collection of the match objects
+                that are created when you use the -match operator. This isn't as flexible nor as
+                robust as using [regex]::Matches(), but it's handy in a pinch.
+                
+                In this case, you'll see that the "sh" in "shears" weren't part of the match. The
+                pattern "zero or more p's, followed by e, a, r, s," is found in the string, even
+                though there are parts of the string that aren't relevant to that matching effort.
+    
+                $matches[0] is always your most recent match.
+            #>
+    
+            '____' | Should -Be $matches[0]
+        }
+
+        It 'is not just an array of text' {
+            # What else do you think $matches might contain?
+            '____' | Should -Be $matches.GetType().FullName
+        }
+    }
+
+    Context '+' {
+
+        It 'specifies 1 or more of something' {
+            <#
+                The + symbol is a lot like the *, but instead of matching zero or more, it
+                matches one or more.
+            #>
+    
+            '____' -match 'p+ickles' | Should -BeTrue
+        }
+
+        It 'needs the string to start with p' {
+            # Change the value of $pickleMatch to pass this test
+            $pickleMatch = 'pickles'
+            $pickleMatch = '____'
+            $pickleMatch -match 'p+ickles' | Should -BeFalse
+        }
+    }
+
+    Context '?' {
+
+        It 'specifies 0 or 1 of something' {
+            # The ? matches "zero or one" of something.
+            $____ | Should -Be ('flying through the sky' -match 'f+lying')
+        }
+        
+        It 'does not mind matching 0 of something' {
+            $____ | Should -Be ('floating away' -match 'b+oat')
+        }
+    }
+
+    Context 'Greediness and Laziness' {
+
+        It 'matches a little or matches a lot' {
+            <#
+            A number of the quantifiers have two versions: A greedy version which tries to match an
+            element as many times as possible, and a non-greedy (or lazy) version which tries to
+            match an element as few times as possible. You can turn a greedy quantifier into a lazy
+            quantifier by simply adding a ?.
+            #>
+
+            $lazyPattern = 'p+?'
+            $lazyMatch = [regex]::Matches('pineapple', $lazyPattern)[0].Value
+            __ | Should -Be $lazyMatch.Length
+        }
+
+        It 'is a hungry bugger' {
```

Comment:
```suggestion
        It 'is greedy by default' {
```

It might also be better if we move this before the above `It` block, and then move the comment there to just under the `Context` declaration.